### PR TITLE
Ensure tasks are assigned to a container

### PR DESCRIPTION
**Pull request by maxfierke**
This fixes an issue with service injection inside
of encapsulated tasks on Ember 3.13+ by ensuring
tasks that are children of other tasks are
assigned a container to use for service injection.

### CODE REVIEW

File: tests/dummy/app/task-injection-test/controller.js [hunk 1]
```diff
@@ -7,18 +7,17 @@ export default Controller.extend({
 
   myTask: task({
     fun: service(),
-    perform: function * () {
+    *perform() {
```
Comment (Turbo87) [inline tests/dummy/app/task-injection-test/controller.js]: seems unrelated?

**Comment by Turbo87** [comment]
@maxfierke is the test suite failing without this patch? otherwise it might make sense to create a test for it that verifies that this works correctly with both old and new Ember versions

**Comment by maxfierke** [comment]
@Turbo87 yeah, the build fails on master for `ember-beta` and `ember-canary` without this

File: tests/dummy/app/task-injection-test/controller.js (hunk 1 above)
Comment (maxfierke) [reply tests/dummy/app/task-injection-test/controller.js]: unrelated to the fix but _cleaner_. Didn't feel like it warranted omission 😛
